sdk/log/record.go

```diff
@@ -86,6 +86,8 @@ type Record struct {
 
 	attributeValueLengthLimit int
 	attributeCountLimit       int
+
+	noCmp [0]func() //nolint: unused  // This is indeed used.
```

Comment:
This is not necessary, but may be good for maintaining backward compatiblity.
Should we do the same for `Record`, `KeyValue`, Value` in `go.opentelemetry.io/otel/log`?

Reply:
> Should we do the same for `Record`, `KeyValue`, `Value` in `go.opentelemetry.io/otel/log`?

SGTM :+1:. Please feel free to open a PR.

Reply:
I created https://github.com/open-telemetry/opentelemetry-go/issues/5696